### ISSUE TITLE
When typing in editor, macOS Accessibility Keyboard's word suggestion fails

### ISSUE THREAD

**Issue by blakewatson** [Microsoft/vscode#45096]
### Issue Type
Bug

### Description

The Accessibility Keyboard in macOS High Sierra has a word prediction feature that allows the user to type a few characters and get suggestions on possible dictionary words. This feature is helpful for oncreen keyboard users because it saves precious key strokes/clicks.

When typing in VS Code's editor, the suggestion lists resets itself on each key stroke, offering suggestions only for the last key typed, instead of allowing the user to build up a word.

The behavior can be seen in this GIF.

https://www.blakewatson.com/externalfiles/vscode-accessibility-keyboard.gif

I tested in a SCSS file I had open, as well as in the interactive playground. I also tested with extensions off.

Note that the issue does not happen in the Monaco Editor online demo.

### VS Code Info

VS Code version: Code 1.20.1 (f88bbf9137d24d36d968ea6b2911786bfe103002, 2018-02-13T15:23:28.607Z)
OS version: Darwin x64 17.4.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-4690 CPU @ 3.50GHz (4 x 3500)|
|Load (avg)|2, 2, 2|
|Memory (System)|24.00GB (3.19GB free)|
|Process Argv|/Applications/Visual Studio Code.app/Contents/MacOS/Electron|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (2)</summary>

Extension|Author (truncated)|Version
---|---|---
format-indent|Kas|1.3.0
php-intellisense|fel|2.2.9

(1 theme extensions excluded)

</details>
Reproduces without extensions

**Comment by kieferrm** [comment]
Reproduced on macOS HighSierra with the built-in accessibility keyboard.

**Comment by rebornix** [comment]
Technically it's about how we handle content in the textarea, and we already did optimization for Screen Reader. @blakewatson you can set `editor.accessibilitySupport` to `on` and you should be able to see word based suggestions as expected.

We should see if there is a way to automatically detect Accessibility Keyboard and turn on this optimization without asking users to change the setting.

**Comment by blakewatson** [comment]
Thanks @rebornix and @kieferrm, that worked. What's the tradeoff, if any, to having `editor.accessibilitySupport` on vs not?

**Comment by rebornix** [comment]
@alexandrudima can you help explain what's the side effect of turning on `accessibilitySupport`? Thanks ;)

**Comment by blakewatson** [comment]
I found a dialog that says "Some editor features will have different behavior behavior: e.g. word wrapping, folding, etc."

**Comment by alexandrudima** [comment]
By default, we only leave the last typed character in the `<textarea>` on OSX when we don't detect a screen reader is attached. That covers the cases when long pressing `e` into `é`. We could leave the entire word in the `<textarea>` to cover the Accessibility Keyboard case.

**Comment by blakewatson** [comment]
@alexandrudima Ah, yeah from my perspective as a user that would be a great solution if doable—then I could keep access to text wrapping and whatever other features get restricted when `accessibilitySupport` is on.

**Comment by alexandrudima** [comment]
To the verifier:
Please check that the "long press" input method still works. e.g. `e` ===> `è`, etc.

Before:

![kapture 2018-03-29 at 12 37 43](https://user-images.githubusercontent.com/5047891/38084654-05cce852-334e-11e8-813d-3e9071f2e813.gif)

After:

![kapture 2018-03-29 at 12 43 35](https://user-images.githubusercontent.com/5047891/38084882-d8545e54-334e-11e8-8472-ae698e65bb0b.gif)
